web_widget_numeric_step/static/src/js/numeric_step.js

```diff
@@ -0,0 +1,239 @@
+         * and can accept alphanumeric characters.
+         * The bad news is that require implement all good 'number' type
+         * features like the minus and plus buttons, steps, min and max...
+         * Perhaps in a near future this can be improved to have the best of
+         * two types without hacky developments.
+         *
+         * @override
+         */
+        _render: function () {
+            this.nodeOptions.type = "tel";
+            this._super.apply(this, arguments);
+        },
+
+        /**
+         * Select the properly widget input
+         *
+         * @override
+         */
+        _renderEdit: function () {
+            this._prepareInput(this.$el.find('input.input_numeric_step'));
+        },
+
+        /**
+         * Increase/Decrease widget input value
+         *
+         * @param {String} mode
+         */
+        _doStep: function (mode) {
+            var cval = Number(this.$input.val());
+            if (_.isNaN(cval)) {
+                return;
+            }
+            if (mode === 'plus') {
+                cval += Number(this.nodeOptions.step) || 1;
+            } else if (mode === 'minus') {
+                cval -= Number(this.nodeOptions.step) || 1;
+            }
+            this.$input.val(this._sanitizeNumberValue(cval));
+            this.isDirty = true;
+            this._doDebouncedAction();
+        },
+
+        // Handle Events
+        _onStepClick: function (ev) {
+            if (!this._autoStep) {
+                var mode = ev.target.dataset.mode;
+                this._doStep(mode);
+            }
+            this._autoStep = false;
+        },
+
+        _onStepMouseDown: function (ev) {
+            this._interval = setTimeout(
+                $.proxy(this._whileMouseDown, this, ev), this._click_delay);
+        },
+
+        _onMouseUp: function () {
+            clearTimeout(this._interval);
+            this._interval = false;
+            this._click_delay = this.DEF_CLICK_DELAY;
+        },
+
+        _whileMouseDown: function (ev) {
+            this._autoStep = true;
+            var mode = ev.target.dataset.mode;
+            this._doStep(mode);
+            if (this._click_delay > this.MIN_DELAY) {
+                this._click_delay -= this.SUBSTRACT_DELAY_STEP;
+            }
+
+            this._onStepMouseDown(ev);
+        },
+
+        /**
+         * Enable mouse wheel support
+         *
+         * @param {WheelEvent} ev
+         */
+        _onWheel: function (ev) {
+            ev.preventDefault();
+            if (ev.originalEvent.deltaY > 0) {
+                this._doStep('minus');
+            } else {
+                this._doStep('plus');
+            }
+        },
+
+        /**
+         * Enable keyboard arrows support
+         *
+         * @param {KeyEvent} ev
+         */
+        _onKeyDown: function (ev) {
+            if (ev.keyCode === KeyCodes.ARROW_UP) {
```

Comment:
Possibly you can use this instead:

```suggestion
            if (ev.key === "ArrowUp") {
```